File: MidTerm/CSVExportAnalytics.js
const fs = require('fs');
const path = require('path');
const readline = require('readline');

class CSVAnalytics {
    constructor() {
        this.csvData = [];
        this.outputPath = path.join(__dirname, 'summary_report.csv');
    }
    
    loadCSV(filename) {
        return new Promise((resolve, reject) => {
            const filePath = path.join(__dirname, filename);
            const stream = fs.createReadStream(filePath);
            const rl = readline.createInterface({
                input: stream,
                crlfDelay: Infinity
            });
            
            let isHeader = true;
            
            rl.on('line', (line) => {
                if (isHeader) {
                    isHeader = false;
                    return;
                }
                
                const parts = line.split(',');
                if (parts.length >= 8) {
                    this.csvData.push({
                        title: parts[1],
                        console: parts[2],
                        genre: parts[3],
                        publisher: parts[4],
                        sales: parts[7]
                    });
                }
            });
            
            rl.on('close', () => {
                console.log(`✓ CSV file loaded successfully!`);
                console.log(`Total Records: ${this.csvData.length}\n`);
                resolve();
            });
            
            rl.on('error', (err) => {
                console.error(`✗ Error loading file: ${err.message}`);
                reject(err);
            });
        });
    }
    
    exportSummary() {
        let csvContent = 'Game Title,Console,Genre,Publisher,Total Sales\n';
        
        this.csvData.forEach(record => {
            csvContent += `${record.title},${record.console},${record.genre},${record.publisher},${record.sales}\n`;
        });
        
        fs.writeFile(this.outputPath, csvContent, (err) => {
            if (err) {
                console.error(`✗ Error exporting file: ${err.message}`);
            } else {
                console.log(`✓ CSV file 'summary_report.csv' exported successfully!`);
                console.log(`Location: ${this.outputPath}`);
            }
        });
    }
    
    displaySample() {
        console.log('Sample Data (First 5 records):');
        console.log('==============================');
        this.csvData.slice(0, 5).forEach(record => {
            console.log(`${record.title} - ${record.console} - $${record.sales}M`);
        });
    }
}

async function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    rl.question('Enter CSV filename: ', async (filename) => {
        const analytics = new CSVAnalytics();
        
        try {
            await analytics.loadCSV(filename);
            analytics.displaySample();
            analytics.exportSummary();
        } catch (error) {
            console.error('Failed to process file');
        }
        
        rl.close();
    });
}

main();

File: MidTerm/CSVExportAnalytics.java
import javax.swing.*;
import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.*;

public class CSVExportAnalytics extends JFrame {
    private JTextArea reportArea;
    private JButton loadButton, exportButton;
    
    public CSVExportAnalytics() {
        setTitle("CSV Export Analytics Report");
        setSize(600, 500);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setLocationRelativeTo(null);
        
        JPanel panel = new JPanel();
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
        
        reportArea = new JTextArea(20, 50);
        reportArea.setEditable(false);
        JScrollPane scrollPane = new JScrollPane(reportArea);
        
        loadButton = new JButton("Load CSV File");
        loadButton.addActionListener(e -> loadCSV());
        
        exportButton = new JButton("Export Summary Report");
        exportButton.addActionListener(e -> exportSummary());
        exportButton.setEnabled(false);
        
        panel.add(new JLabel("CSV Analytics Data:"));
        panel.add(scrollPane);
        panel.add(loadButton);
        panel.add(exportButton);
        
        add(panel);
        setVisible(true);
    }
    
    private List<String[]> csvData = new ArrayList<>();
    
    private void loadCSV() {
        JFileChooser fileChooser = new JFileChooser();
        int result = fileChooser.showOpenDialog(this);
        
        if (result == JFileChooser.APPROVE_OPTION) {
            File file = fileChooser.getSelectedFile();
            try {
                csvData.clear();
                List<String> lines = Files.readAllLines(Paths.get(file.getAbsolutePath()));
                
                reportArea.setText("Loading CSV File: " + file.getName() + "\n\n");
                reportArea.append("Total Records: " + (lines.size() - 1) + "\n\n");
                reportArea.append("Sample Data:\n");
                reportArea.append("============\n");
                
                for (int i = 0; i < Math.min(10, lines.size()); i++) {
                    String[] parts = lines.get(i).split(",");
                    csvData.add(parts);
                    reportArea.append(lines.get(i) + "\n");
                }
                
                exportButton.setEnabled(true);
                JOptionPane.showMessageDialog(this, "✓ CSV file loaded successfully!");
                
            } catch (IOException ex) {
                JOptionPane.showMessageDialog(this, "✗ Error loading file: " + ex.getMessage());
            }
        }
    }
    
    private void exportSummary() {
        if (csvData.isEmpty()) {
            JOptionPane.showMessageDialog(this, "✗ No data to export!");
            return;
        }
        
        try {
            String outputPath = "c:\\Users\\Administrator\\Desktop\\HADAP\\Prog2-9307-AY225-HADAP\\MidTerm\\summary_report.csv";
            FileWriter writer = new FileWriter(outputPath);
            
            writer.append("Game Title,Console,Genre,Publisher,Total Sales\n");
            
            for (String[] row : csvData) {
                if (row.length >= 5) {
                    writer.append(row[1]).append(",")
                          .append(row[2]).append(",")
                          .append(row[3]).append(",")
                          .append(row[4]).append(",")
                          .append(row[7]).append("\n");
                }
            }
            
            writer.flush();
            writer.close();
            
            reportArea.append("\n✓ Summary report exported to: summary